### PR TITLE
feat(vm): support wasm call host http API

### PR DESCRIPTION
**Pull request by huangzhiran**
https://iotex.larksuite.com/docx/IAODdYkwFowx03x8HBtu8YUcsKb

### CODE REVIEW

File: pkg/modules/vm/api/server.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+import (
+	"net/http"
+	"net/http/httptest"
+)
+
+var DefaultServer = NewServer()
+
+type Server struct {
+	srv *http.ServeMux
```
Comment (Liuhaai) [inline pkg/modules/vm/api/server.go:11]: should a real server be setup?

**Comment by Liuhaai** [comment]
The entire idea looks good to me. Wait for the confirmation of WASM adapation (AS, Rust, Golang) for the next step.

File: pkg/modules/vm/api/server.go (hunk 1 above)
Comment (huangzhiran) [reply pkg/modules/vm/api/server.go:11]: it not a real server. will not listen a port real

File: pkg/modules/vm/api/server.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package api
+
+import (
+	"net/http"
+	"net/http/httptest"
+)
+
+var DefaultServer = NewServer()
```
Comment (CoderZhi) [inline pkg/modules/vm/api/server.go:8]: delete and move initialization to server side

File: pkg/modules/vm/wasmtime/instance_exports.go [hunk 3]
```diff
@@ -50,6 +55,7 @@ func NewExportFuncs(ctx context.Context, rt *Runtime) (*ExportFuncs, error) {
 		kvs: wasm.MustKVStoreFromContext(ctx),
 		log: wasm.MustLoggerFromContext(ctx),
 		ctx: ctx,
+		srv: api.DefaultServer,
```
Comment (CoderZhi) [inline pkg/modules/vm/wasmtime/instance_exports.go:58]: get from context

**Review by CoderZhi — COMMENTED** [review]
as discussed offline

File: pkg/modules/vm/api/server.go (hunk 1 above)
Comment (Liuhaai) [reply pkg/modules/vm/api/server.go:9]: good to know

**Review by Liuhaai — APPROVED** [review]

**Pull request merged**